### PR TITLE
Nan to null

### PR DESCRIPTION
**Pull request by wabinyai**
**_WHAT DOES THIS PR DO?_**
Python JSON encoder converts NaNs to null 

- [ ] Item 1
- [ ] Item 2

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - []
- GitHub issues
    - Closes #<enter GitHub issue number here>

**_HOW DO I TEST OUT THIS PR?_**

- [x] Analytics: [Link to Analytics README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/analytics#readme)
- [ ] Auth Service: [Link to Auth Service README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/auth-service#readme)
- [ ] Calibrate: [Link to Calibrate README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/calibrate#readme)
- [ ] Data Proxy: [Link to Data Proxy README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/data-proxy#readme)
- [ ] Device Monitoring: [Link to Device Monitoring README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-monitoring#readme)
- [ ] Device Registry: [Link to Device Registry README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-registry#readme)
- [ ] Device Status: [Link to Device Status README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-status#readme)
- [ ] Device Uptime: [Link to Device Uptime README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/device-uptime#readme)
- [ ] Exceedances: [Link to Exceedances README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/exceedances#readme)
- [ ] Firebase: [Link to Firebase README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/firebase)
- [ ] GP Model: [Link to GP Model README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/gp-model#readme)
- [ ] Incentives: [Link to Incentives README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/incentives#readme)
- [ ] Kafka Connectors: [Link to Kafka Connectors](https://github.com/airqo-platform/AirQo-api/tree/staging/src/kafka-connectors)
- [ ] Locate: [Link to Locate README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/locate#readme)
- [ ] Metadata: [Link to Metadata README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/meta-data#readme)
- [ ] Predict: [Link to Predict README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/predict#readme)
- [ ] Spatial: [Link to Spatial](https://github.com/airqo-platform/AirQo-api/tree/staging/src/spatial)
- [ ] View: [Link to View README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/view#readme)
- [ ] Workflows: [Link to Workflows README](https://github.com/airqo-platform/AirQo-api/tree/staging/src/workflows#readme)

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [ ] Endpoint 1
- [ ] Endpoint 2

**_ARE THERE ANY RELATED PRs?_**

- [ ] Related PR 1
- [ ] Related PR 2

### CODE REVIEW

**Review by Baalmart — APPROVED** [review]

File: src/analytics/api/models/base/data_processing.py [hunk 1]
```diff
@@ -89,6 +90,16 @@ def air_quality_data():
 
                 }
             }
+            def replace_nan_with_null(obj):
```
Comment (Baalmart) [inline src/analytics/api/models/base/data_processing.py:93]: Thanks @wabinyai .....already merged but leaving a performance optimisation consideration.....

The function uses list comprehensions and dictionary comprehensions for lists and dictionaries, which is generally efficient. However, for very large nested structures, the recursive nature of the function could lead to a stack overflow error due to Python's recursion limit. This is unlikely to be an issue for most practical use cases but is something to be aware of.

Comment (wabinyai) [reply src/analytics/api/models/base/data_processing.py:93]: Thanks .

Comment (wabinyai) [reply src/analytics/api/models/base/data_processing.py:93]: Noted with thanks.

Comment (Baalmart) [reply src/analytics/api/models/base/data_processing.py:93]: @wabinyai  , additionally...
The function correctly handles lists and dictionaries by recursively calling itself on their elements. However, it does not explicitly handle other data types that might contain NaN values, such as numpy arrays or pandas DataFrames. If the input object contains such data types, the function will not replace NaN values within them.

**Pull request merged**